runtime/parachains/src/disputes/tests.rs

```diff
@@ -888,12 +686,122 @@ fn test_freeze_provided_against_supermajority_for_included() {
+	fn generate_dispute_statement_set_and_run_to_block() -> DisputeStatementSet {
+		// 7 validators needed for byzantine threshold of 2.
+		let v0 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v1 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v2 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v3 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v4 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v5 = <ValidatorId as CryptoType>::Pair::generate().0;
+		let v6 = <ValidatorId as CryptoType>::Pair::generate().0;
+
+		// Mapping between key pair and `ValidatorIndex`
+		// v0 -> 0
+		// v1 -> 3
+		// v2 -> 6
+		// v3 -> 5
+		// v4 -> 1
+		// v5 -> 4
+		// v6 -> 2
+
+		run_to_block(6, |b| {
+			// a new session at each block
+			Some((
+				true,
+				b,
+				vec![
+					(&0, v0.public()),
+					(&1, v1.public()),
+					(&2, v2.public()),
+					(&3, v3.public()),
+					(&4, v4.public()),
+					(&5, v5.public()),
+					(&6, v6.public()),
+				],
+				Some(vec![
+					(&0, v0.public()),
+					(&1, v1.public()),
+					(&2, v2.public()),
+					(&3, v3.public()),
+					(&4, v4.public()),
+					(&5, v5.public()),
+					(&6, v6.public()),
+				]),
+			))
+		});
+
+		let candidate_hash = CandidateHash(sp_core::H256::repeat_byte(1));
+
+		// v0 votes for 4, v1 votes against 4.
+		DisputeStatementSet {
+			candidate_hash: candidate_hash.clone(),
+			session: 4,
+			statements: vec![
+				(
+					DisputeStatement::Valid(ValidDisputeStatementKind::Explicit),
+					ValidatorIndex(0),
+					v0.sign(
+						&ExplicitDisputeStatement {
+							valid: true,
+							candidate_hash: candidate_hash.clone(),
+							session: 4,
+						}
+						.signing_payload(),
+					),
+				),
+				(
+					DisputeStatement::Invalid(InvalidDisputeStatementKind::Explicit),
+					ValidatorIndex(3),
+					v1.sign(
+						&ExplicitDisputeStatement {
+							valid: false,
+							candidate_hash: candidate_hash.clone(),
+							session: 4,
+						}
+						.signing_payload(),
+					),
+				),
+			],
+		}
+	}
+	#[test]
+	fn test_unconfirmed_are_ignored() {
+		new_test_ext(Default::default()).execute_with(|| {
+			let stmts = vec![generate_dispute_statement_set_and_run_to_block()];
+			let stmts = filter_dispute_set(stmts);
+
+			// Not confirmed => should be filtered out
+			assert_ok!(Pallet::<Test>::process_checked_multi_dispute_data(stmts), vec![],);
+		});
+	}
+
+	#[test]
+	fn test_unconfirmed_disputes_cause_block_import_error() {
+		new_test_ext(Default::default()).execute_with(|| {
+
+		let stmts = generate_dispute_statement_set_and_run_to_block();
+		let stmts = vec![CheckedDisputeStatementSet::unchecked_from_unchecked(stmts)];
+
+		assert_matches!(
+			Pallet::<Test>::process_checked_multi_dispute_data(stmts),
+			Err(DispatchError::Module(ModuleError{index: _, error: _, message})) => assert_eq!(message, Some("UnconfirmedDispute"))
```

Comment:
nit: To be thorough you could check that you hit the error with 2 votes but not with 3.